cli/commands/stack/stack.go

```diff
@@ -137,21 +137,59 @@ func FilterOutputs(outputs cty.Value, index string) cty.Value {
 	return nested
 }
 
-// RunClean cleans the stack directory
+// RunClean recursively removes all stack directories under the specified WorkingDir.
 func RunClean(ctx context.Context, opts *options.TerragruntOptions) error {
-	baseDir := filepath.Join(opts.WorkingDir, stackDir)
-	err := telemetry.TelemeterFromContext(ctx).Collect(ctx, "stack_clean", map[string]any{
+	telemeter := telemetry.TelemeterFromContext(ctx)
+	err := telemeter.Collect(ctx, "stack_clean", map[string]any{
 		"stack_config_path": opts.TerragruntStackConfigPath,
 		"working_dir":       opts.WorkingDir,
 	}, func(ctx context.Context) error {
-		opts.Logger.Debugf("Cleaning stack directory: %s", baseDir)
-		err := os.RemoveAll(baseDir)
+		var (
+			firstErr error
```

Comment:
Why do we only care about the first error instead of collecting all the errors and returning errors.join?

Reply:
Added collection of all errors